js/class/ToDoApplication.js

```diff
@@ -0,0 +1,136 @@
+        const ulList = document.querySelector('#todo-list');
+        let liTemplate = `<li id="${newItem.id}" class="${newItem.isCompleted ? 'completed' : ''}">
+                        <div class="view">
+                        <input class="toggle" type="checkbox" ${newItem.isCompleted ? "checked" : ''} />
+                        <label class="label">${newItem.text}</label>
+                        <button class="destroy"></button>
+                        </div>
+                        <input class="edit" value="${newItem.text}" />
+                    </li>`;
+        ulList.innerHTML += liTemplate;
+    }
+    deleteTodoItem(){
+        let list = document.querySelector('#todo-list');
+        let btn = document.querySelectorAll('.destroy');
+        let parent,liId;
+        for(let i =0; i < btn.length; i++){
+            btn[i].onclick = (e) => {
+                parent = e.target.parentNode.parentNode;
+                liId = parent.id;
+                list.removeChild(parent);
+                this.toDo.deleteItem(liId);
+                this.countToDos();
+            };
+        }
+        
+    }
+    setComplete(){
+        let checkBoxList = document.querySelectorAll('.toggle');
+        checkBoxList.forEach((check) => {
+            check.addEventListener('change',(e) => {
+                let active = e.target.parentNode.parentNode;
+                let activeId = active.id;
+                if(e.target.checked){
+                    e.target.setAttribute('checked','checked');
+                    active.classList.value = 'completed';
+                    this.toDo.isCompleted(activeId,true);
+                    this.filterToDos();
+                }else{
+                    e.target.setAttribute('checked',false);
+                    active.classList.value = '';
+                    this.toDo.isCompleted(activeId,false);
+                    this.filterToDos();
+                }
+            });
+
+        });
+    }
+    countToDos(){
+        let count = this.toDo.items.length;
+        let view = document.querySelector('main span.todo-count strong');
+        view.innerText = count;
+    }
+    filterToDos(){
+        let alist = document.querySelectorAll('a');
+        alist.forEach((filterA) => {
+            filterA.addEventListener('click',(e) => {
+                let beforeSelected = document.querySelector('.selected');
+                beforeSelected.classList.remove('selected');
+                filterA.classList.add('selected');
+                let filterName = filterA.classList[0];
+                let filter;
+                if(filterName == 'all'){
+                    filter = this.toDo.items;
+                }else{
+                    filter = this.toDo.filterItem(filterName);
+                }
+                let list = document.querySelector('#todo-list');
+                list.innerHTML = '';
+                this.loadTodos(filter);
+            });
+        });
+
+    }
```

Comment:
```suggestion
    get filteredItems () {
      const { filterName, toDo: { items } } = this;
      return items.filter(({ isCompleted }) => 
        (filterName === 'all') ||
        (filterName === 'completed' && isCompleted) ||
        (filterName === 'active' && !isCompleted));
    }

    filterToDos(){
        document.querySelectorAll('a').forEach(filterA => {
            filterA.addEventListener('click', e => {
                e.preventDefault();
                document.querySelector('.selected').classList.remove('selected');
                filterA.classList.add('selected');
                this.filterName = filterA.classList[0];
                this.loadTodos(this.todo.items);
            });
        });

    }
```
이렇게 만든 다음에 filteredItems를 렌더링할 때 사용하면 좋을 것 같아요!